packages/data-service/src/data-service.ts

```diff
@@ -1609,29 +1606,34 @@ export class DataServiceImpl extends EventEmitter implements DataService {
     );
   }
 
-  explain(
+  explainFind(
     ns: string,
     filter: Filter<Document>,
-    options: FindOptions,
-    callback: Callback<Document>
-  ): void {
-    const logop = this._startLogOp(
+    options: FindOptions = {},
+    executionOptions?: ExplainExecuteOptions
+  ): Promise<Document> {
+    const verbosity =
+      executionOptions?.explainVerbosity ||
+      mongodb.ExplainVerbosity.allPlansExecution;
+
+    log.info(
       mongoLogId(1_001_000_046),
+      this._logCtx(),
       'Running find explain',
-      { ns }
+      { ns, verbosity }
+    );
+
+    let cursor: FindCursor;
+    return this.cancellableOperation(
+      async (session?: ClientSession) => {
+        cursor = this.findCursor(ns, filter, { ...options, session });
+        const results = await cursor.explain(verbosity);
+        void cursor.close();
+        return results;
+      },
+      () => cursor?.close(),
+      executionOptions?.abortSignal
     );
-    // @todo thomasr: driver explain() does not yet support verbosity,
-    // once it does, should be passed along from the options object.
-    this._collection(ns, 'CRUD')
-      .find(filter, options)
-      .explain((error, explanation) => {
-        logop(error);
```

Comment:
This now means that the logs won’t contain information about the success or failure of the operation. Do we maybe want to integrate this into `cancellableOperation()` a bit more so that we do log errors?

Reply:
added in [c71d7e8](https://github.com/mongodb-js/compass/pull/3860/commits/c71d7e8b757e2fb981080aaf6ac26ad711601bb5)